Panic when trying to copy file from another target when the file does not exist

### What went wrong?

The following Earthfile will cause earthly to panic:

```Dockerfile
VERSION 0.8

compile:
  FROM alpine:3
  SAVE ARTIFACT /doesnotexist

build:
  FROM alpine:3
  COPY +compile/doesnotexist .
```

Repro with command:


Output:
```
$ earthly +build
 Init 🚀
————————————————————————————————————————————————————————————————————————————————

           buildkitd | Found buildkit daemon as docker container (earthly-buildkitd)

Streaming logs to https://cloud.earthly.dev/builds/a24dc822-06ba-484c-89bc-e388665ab9b3

 Build 🔧
————————————————————————————————————————————————————————————————————————————————

              logbus | Setting organization "jahands" and project ""
              +build | --> FROM +base
            alpine:3 | --> Load metadata alpine:3 linux/arm64
       registry auth | logged into registry-1.docker.io with username jhands (using credentials from /user/std/registry/registry-1.docker.io/)
            +compile | --> FROM +base
            +compile | --> FROM alpine:3
            +compile | [----------] 100% FROM alpine:3
panic: runtime error: index out of range [2] with length 0

goroutine 133 [running]:
github.com/earthly/earthly/logbus/solvermon.formatErrorMessage({0x140003c0780, 0x74}, {0x14000118420, 0x20}, 0x1, 0x3, 0x0)
        github.com/earthly/earthly/logbus/solvermon/vertexmon.go:120 +0x4f0
github.com/earthly/earthly/logbus/solvermon.(*vertexMonitor).parseError(0x1400091d900)
        github.com/earthly/earthly/logbus/solvermon/vertexmon.go:154 +0xc8
github.com/earthly/earthly/logbus/solvermon.(*SolverMonitor).handleBuildkitStatus(0x1400032d580, {0x14000813eb8?, 0xa?}, 0x140009982a0)
        github.com/earthly/earthly/logbus/solvermon/solvermon.go:140 +0x5c0
github.com/earthly/earthly/logbus/solvermon.(*SolverMonitor).MonitorProgress(0x1400032d580, {0x103696498, 0x140000e6b40}, 0x1400062e2a0)
        github.com/earthly/earthly/logbus/solvermon/solvermon.go:62 +0x154
github.com/earthly/earthly/builder.(*solver).buildMainMulti.func2()
        github.com/earthly/earthly/builder/solver.go:72 +0x30
golang.org/x/sync/errgroup.(*Group).Go.func1()
        golang.org/x/sync@v0.6.0/errgroup/errgroup.go:78 +0x58
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1
        golang.org/x/sync@v0.6.0/errgroup/errgroup.go:75 +0x98
```

### What should have happened?

Shown a useful error message about the file not being found (maybe even have SAVE ARTIFACT fail if it's not found?)

### What earthly version?

$ earthly --version
earthly version v0.8.14 6b41f8409d7ffef0d25072c2c04250074b6e3c72 darwin/arm64; macOS 14.5 homebrew

### Buildkit Logs

_No response_

### Other Helpful Information

This issue was introduced in #4180 here: https://github.com/earthly/earthly/blob/main/logbus/solvermon/vertexmon.go#L120
We're trying to apply a regex `^failed to calculate checksum of ref ([^ ]*): (.*)$`
To a string like this:
```
 internalfailed to calculate checksum of ref p4gz72iufvk3t1nsqq07p9sim::m4m7o7gui4zuuoy9vynbrzx8f: "/doesnotexist": not found
```

--- a/logbus/solvermon/vertexmon.go
+++ b/logbus/solvermon/vertexmon.go
@@ -63,7 +63,7 @@ func getExitCode(errString string) (int, error) {
 	return 0, errNoExitCode
 }
 
-var reErrNotFound = regexp.MustCompile(`^failed to calculate checksum of ref ([^ ]*): (.*)$`)
+var reErrNotFound = regexp.MustCompile(`^\s*(internal)?failed to calculate checksum of ref ([^ ]::[^ ]*|[^ ]*): (.*)\s*$`)
 var reHint = regexp.MustCompile(`^(?P<msg>.+?):Hint: .+`)
 
 // determineFatalErrorType returns logstream.FailureType
@@ -114,10 +114,14 @@ func formatErrorMessage(errString, operation string, internal bool, fatalErrorTy
 				"      did not complete successfully. Exit code %d", internalStr, operation, exitCode)
 	case logstream.FailureType_FAILURE_TYPE_FILE_NOT_FOUND:
 		m := reErrNotFound.FindStringSubmatch(errString)
+		reason := fmt.Sprintf("unable to parse file_not_found error:%s", errString)
+		if len(m) > 2 {
+			reason = m[3]
+		}
 		return fmt.Sprintf(
 			"      The%s command\n"+
 				"          %s\n"+
-				"      failed: %s", internalStr, operation, m[2])
+				"      failed: %s", internalStr, operation, reason)
 	case logstream.FailureType_FAILURE_TYPE_GIT:
 		gitStdErr, shorterErr, ok := errutil.ExtractEarthlyGitStdErr(errString)
 		if ok {

--- a/logbus/solvermon/vertexmon_test.go
+++ b/logbus/solvermon/vertexmon_test.go
@@ -5,6 +5,7 @@ import (
 	"testing"
 
 	"github.com/earthly/cloud-api/logstream"
+	"github.com/stretchr/testify/assert"
 )
 
 func TestGetExitCode(t *testing.T) {
@@ -95,6 +96,22 @@ func TestDetermineFatalErrorType(t *testing.T) {
 			expectedFatal: true,
 		},
 		{
+			name:          "file not found (internal)",
+			errString:     "internalfailed to calculate checksum of ref foo: bar",
+			exitCode:      0,
+			parseErr:      nil,
+			expectedType:  logstream.FailureType_FAILURE_TYPE_FILE_NOT_FOUND,
+			expectedFatal: true,
+		},
+		{
+			name:          "file not found (internal with space)",
+			errString:     " internalfailed to calculate checksum of ref foo: bar",
+			exitCode:      0,
+			parseErr:      nil,
+			expectedType:  logstream.FailureType_FAILURE_TYPE_FILE_NOT_FOUND,
+			expectedFatal: true,
+		},
+		{
 			name:          "git error",
 			errString:     "EARTHLY_GIT_STDERR: Z2l0IC1jI...",
 			parseErr:      nil,
@@ -125,6 +142,50 @@ func TestDetermineFatalErrorType(t *testing.T) {
 			}
 			if fatal != tt.expectedFatal {
 				t.Errorf("determineFatalErrorType(%q, %d) = %v, want %v", tt.errString, tt.exitCode, fatal, tt.expectedFatal)
+			}
+		})
+	}
+}
+
+func TestReErrNotFound(t *testing.T) {
+	tests := []struct {
+		name      string
+		errString string
+		expected  []string
+	}{
+		{
+			name:      "simple",
+			errString: "failed to calculate checksum of ref foo: bar",
+			expected:  []string{"", "foo", "bar"},
+		},
+		{
+			name:      "simple (internal)",
+			errString: "internalfailed to calculate checksum of ref foo: bar",
+			expected:  []string{"internal", "foo", "bar"},
+		},
+		{
+			name:      "simple (internal with space)",
+			errString: " internalfailed to calculate checksum of ref foo: bar",
+			expected:  []string{"internal", "foo", "bar"},
+		},
+		{
+			name:      "complex",
+			errString: ` failed to calculate checksum of ref p4gz72iufvk3t1nsqq07p9sim::m4m7o7gui4zuuoy9vynbrzx8f: "/doesnotexist": not found`,
+			expected:  []string{"", "p4gz72iufvk3t1nsqq07p9sim::m4m7o7gui4zuuoy9vynbrzx8f", `"/doesnotexist": not found`},
+		},
+		{
+			name:      "complex (internal)",
+			errString: ` internalfailed to calculate checksum of ref p4gz72iufvk3t1nsqq07p9sim::m4m7o7gui4zuuoy9vynbrzx8f: "/doesnotexist": not found`,
+			expected:  []string{"internal", "p4gz72iufvk3t1nsqq07p9sim::m4m7o7gui4zuuoy9vynbrzx8f", `"/doesnotexist": not found`},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			match := reErrNotFound.FindStringSubmatch(tt.errString)
+
+			if len(match) == 0 || !assert.ElementsMatch(t, match[1:], tt.expected) {
+				t.Errorf("reErrNotFound.FindStringSubmatch(%s) = %v, want %v", tt.errString, match, tt.expected)
 			}
 		})
 	}

